perun/collect/complexity/run.py

```diff
@@ -39,174 +38,206 @@
+    """ Assembles the SystemTap script according to input parameters and collection strategy
 
-    Arguments:
-        kwargs(dict): dictionary containing the configuration settings for the complexity collector
-
-    Returns:
-        tuple: int as a status code, nonzero values for errors
-               string as a status message, mainly for error states
-               dict of modified kwargs with bin value representing the executable
+    :param dict kwargs: dictionary containing the configuration settings for the collector
+    :returns: tuple (int as a status code, nonzero values for errors,
+                    string as a status message, mainly for error states,
+                    dict of modified kwargs with 'script' value as a path to the script file)
     """
     try:
-        # Extract several keywords to local variables
-        target_dir, files, rules = kwargs['target_dir'], kwargs['files'], kwargs['rules']
-        # Create the configuration cmake and build the configuration executable
-        print('Building the configuration executable...')
-        cmake_path = makefiles.create_config_cmake(target_dir, files)
-        exec_path = makefiles.build_executable(cmake_path, makefiles.CMAKE_CONFIG_TARGET)
-        print('Build complete.')
-        # Extract some configuration data using the configuration executable
-        print('Extracting the configuration...')
-        function_sym = symbols.extract_symbols(exec_path)
-        include_list, exclude_list, runtime_filter = symbols.filter_symbols(function_sym, rules)
-        # Create the collector cmake and build the collector executable
-        print('Building the collector executable...')
-        cmake_path = makefiles.create_collector_cmake(target_dir, files, exclude_list)
-        exec_path = makefiles.build_executable(cmake_path, makefiles.CMAKE_COLLECT_TARGET)
-        print('Build complete.\n')
-        # Create the internal configuration file
-        configurator.create_runtime_config(exec_path, runtime_filter, include_list, kwargs)
-
-        kwargs['cmd'] = exec_path
+        print('Starting the pre-processing phase... ', end='')
+
+        # Check the command (must be file)
+        if not os.path.isfile(kwargs['cmd']):
+            raise ValueError('The command argument (-c) is not a file.')
+
+        # Assemble script according to the parameters
+        kwargs['cmd'], kwargs['cmd_dir'], kwargs['cmd_base'] = _get_path_dir_file(kwargs['cmd'])
+        kwargs['script'] = strategy.assemble_script(**kwargs)
+
+        print('Done.\n')
         return 0, _COLLECTOR_STATUS_MSG[0], dict(kwargs)
     # The "expected" exception types
     except (OSError, ValueError, subprocess.CalledProcessError,
-            UnicodeError, exceptions.UnexpectedPrototypeSyntaxError) as exception:
-        return 1, repr(exception), kwargs
+            UnicodeError, exceptions.StrategyNotImplemented) as exception:
+        print('Failed.\n')
+        return 1, str(exception), kwargs
```

Comment:
Same as above, internally it would look better for you to use Enum (or maybe rather IntEnum), however I am afraid, that this will need to be changed in higher part of Perun. I will check it out and remodel the api, so some kind of enum can be returned.